openedx/custom/forms/fields.py

```diff
@@ -0,0 +1,81 @@
+from django.core.exceptions import ValidationError
+from django.forms.widgets import CheckboxSelectMultiple
+from django.utils.translation import ugettext_lazy as _
+from multiselectfield import MultiSelectFormField
+
+
+def get_other_values(choices, value):
+    """
+    This function to separate other's value from list of choices
+    :param choices: list of valid choices
+    :param value: list of selected choices including other's value
+    :return: list of other values.
+    """
+    choice_values = [choice[0] for choice in choices]
+    other_values = [val for val in value if not val in choice_values]
+    return other_values
+
+
+class CheckboxSelectMultipleWithOther(CheckboxSelectMultiple):
+    """
+        Widget class to handle other value filed.
+    """
+    other_choice = None
+
+    def create_option(self, name, value, label, selected, index, subindex=None, attrs=None):
+        option = super(CheckboxSelectMultipleWithOther, self).create_option(name, value, label, selected, index,
+                                                                            subindex, attrs)
+        option.update({
+            'value': self.other_choice if value == 'other' else value,
+            'type': 'text' if value == 'other' else self.input_type,
+            'is_other': label == 'Other'
+        })
+        return option
+
+    def optgroups(self, name, value, attrs=None):
+        """Return a list of optgroups for this widget."""
+
+        other_values = get_other_values(self.choices, value)
+
+        OTHER_CHOICE_INDEX = 0
+        other_values = '' if not other_values else other_values.pop(OTHER_CHOICE_INDEX)
+
+        self.other_choice = other_values
+
+        return super(CheckboxSelectMultipleWithOther, self).optgroups(name, value, attrs)
+
+
+class MultiSelectWithOtherFormField(MultiSelectFormField):
+    """
+    Form field class to handle other text input field with in the multiselect field
+    """
+    widget = CheckboxSelectMultipleWithOther
+
+    def __init__(self, other_max_length=None,*args, **kwargs):
+        super(MultiSelectWithOtherFormField, self).__init__(*args, **kwargs)
+        self.other_max_length = other_max_length
+        self.error_messages.update(
+            dict(invalid_length=_('Other field value. %(value)s maximum allowed length violation.')))
+
+    def valid_value(self, value):
+        return len(value) < self.other_max_length
+
+    def validate(self, value):
+        """Validate that the input is a list or tuple."""
+        if self.required and not value:
+            raise ValidationError(self.error_messages['required'], code='required')
+
+        if self.other_max_length is not None:
+
+            other_values = get_other_values(self.choices, value)
+            for val in other_values:
+                if not self.valid_value(val):
+                    raise ValidationError(
+                        self.error_messages['invalid_length'],
```

Comment:
Can we please remove the blank line at #69? It is unnecessary since indentation is well separating the code. It is disturbing code consistency as the above _if_ block does not have such a blank line, neither does any other _if_ block in this code.